h2o-core/src/main/java/water/fvec/RollupStats.java

```diff
@@ -217,6 +208,24 @@ else if (c instanceof C8DChunk)
     return this;
   }
 
+  private long processUUIDchunk(Chunk c, long checksum, long start, long l) {
+    for( int i=c.nextNZ(-1); i< c._len; i=c.nextNZ(i) ) {
+      if( c.isNA(i) ) _naCnt++;
+      else {
+        try {
+          long lo = c.at16l(i), hi = c.at16h(i);
+          if (lo != 0 || hi != 0) _nzCnt++;
+          l = lo ^ 37*hi;
+        } catch (Chunk.WrongType wt) {
```

Comment:
@michalkurka the thing is, if UUID is specified, and non-UUID data are found, they are not stored as UUID, but as NA or something. Then, if no UUID is found at all in the column, the vec is not detected as UUID, hence the clash with declared and detected type. That's what was happening. The code cannot guarantee that data feed is or is not UUID, the code can only figure out whether some or all data are not; and it does it perfectly.